django/db/models.py

```diff
@@ -496,7 +496,8 @@ def get_salt_id(self):
       
     def get_molecular_weight(self, is_authenticated=False):
         if(not self.reagent.is_restricted or is_authenticated):
-            return str(self._molecular_weight).rstrip('0')
+            return ( str(self._molecular_weight).rstrip('0') 
+                        if self._molecular_weight else '' )
```

Comment:
Suggest `if self._molecular_weight is not None` to be more explicit about the rationale.